server/models/Transaction.js

```diff
@@ -158,6 +159,89 @@ module.exports = function(Sequelize, DataTypes) {
           netAmountInGroupCurrency: this.netAmountInGroupCurrency
         };
       }
+    },
+
+    classMethods: {
+      createFromPayload(payload) {
```

Comment:
Should use ES6 destructuring syntax `{user, group, transaction, subscription, paymentMethod}` with appropriate default values (if any) instead of the black-box `payload`.